R-package/R/lgb.restore_handle.R

```diff
@@ -0,0 +1,36 @@
+#' @name lgb.restore_handle
```

Comment:
Since every method on the Booster object in R will call `$restore_handle()`, why is it necessary to have this exported `lgb.restore_handle()` function? What is the reason that someone using `{lightgbm}` would want to "forcibly restore" the handle before calling a method on the `Booster`?

Unless I'm missing something, I think this function should be removed and `$restore_handle()` should be moved into a private method on the `Booster`. That would reduce the size of the public API that users have to think about.

Reply:
Because it takes time to deserialize it so if you want to serve model predictions you'll want to have it ready from the beginning instead of suffering a delay the first time, especially if you use process forking for serving in parallel (which would not save the state so it'd have to deserialize at every call).

Reply:
Ahhhhh ok ok makes sense! Thanks for the explanation. That's something we can add in the docs in the future, I think.

Ok, then I'm good with keeping this function as-is.